rzup/rzup-init.sh

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env bash
+# Copyright 2024 RISC Zero, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+# Example URLs:
+# for a 64-bit Linux system (x86_64):
+# BASE_URL/rzup-x86_64-linux
+# for a 64-bit macOS system (arm64):
+# BASE_URL/rzup-arm64-darwin
+
+# NOTE: Temporary release URL
+RZUP_BINARY_UPDATE_ROOT="${RZUP_BINARY_UPDATE_ROOT:-https://github.com/hmrtn/asset-test/releases/download/test1/}"
+QUIET=no
+
+CARGO_BIN_DIR="${HOME}/.cargo/bin"
+DEPRECATED_RISC0_DIR="${HOME}/.risc0"
+
+usage() {
+  cat <<EOF
+rzup-init
+
+The installer for rzup, from RISC Zero.
+
+Usage: rzup-init.sh [OPTIONS]
+
+Options:
+  -v, --verbose
+          Enable verbose output
+  -q, --quiet
+          Disable progress output
+  -h, --help
+          Print help
+EOF
+}
+
+cleanup() {
+  [ -n "${_dir:-}" ] && [ -d "$_dir" ] && rm -rf "$_dir"
+}
+
+trap cleanup EXIT
+
+main() {
+  downloader --check
+  need_cmd uname
+  need_cmd mktemp
+  need_cmd chmod
+  need_cmd mkdir
+  need_cmd rm
+  need_cmd rmdir
+  need_cmd mv
+
+  check_rust_installed
+  get_architecture
+
+  _arch="$RETVAL"
+  assert_nz "$_arch" "arch"
+
+  _url="${RZUP_BINARY_UPDATE_ROOT}/rzup-${_arch}"
+  _dir="$(ensure mktemp -d)"
+  _file="${_dir}/rzup"
+
+  [ -f "$DEPRECATED_RISC0_DIR/bin/rzup" ] && rm -rf "$DEPRECATED_RISC0_DIR"
```

Comment:
Note, the init script detects the deprecated installed rzup and removes it if it exists